src/basic/socket-util.c

```diff
@@ -900,6 +900,29 @@ bool ifname_valid(const char *p) {
         return true;
 }
 
+bool iflabel_valid(const char *p) {
+
+        if (isempty(p))
+                return false;
+
+        if (strlen(p) >= IFNAMSIZ)
+                return false;
+
+        if (dot_or_dot_dot(p))
+                return false;
+
+        while (*p) {
+                if ((unsigned char) *p >= 127U)
+                        return false;
+
+                if ((unsigned char) *p <= 32U)
```

Comment:
also, if this is only applicable to addresses, it should probably be called address_label_valid(), no?